### ISSUE TITLE
random NoServerException at startup

### ISSUE THREAD

**Issue by asdonation** [nats-io/nats.net.v1#371]
Hi,

**TL;DR** 
Our dotnet client sometimes fails to connect to nats server on startup, and therefore crashes. This repeats, until it stops for no reason.
We need help tracing the issue.

At startup, Our components subscribe to different topics, and if it fails they exit immediately. Sometimes, for unknown reason, our dotnet components fail to create `IEncodedConnection`, which causes them a crash loop.
We added a tcp connectivity test to all ports (4222, 6222, 8222) at startup. That test always succeeds, also when the following exception occurs
```
 ---> NATS.Client.NATSNoServersException: Unable to connect to a server.
   at NATS.Client.Connection.connect()
   at NATS.Client.ConnectionFactory.CreateEncodedConnection(Options opts)
```

We use a basic configuration options, modifying the `AllowReconnect, ReconnectWait, MaxReconnect=-1` with forgiving timeouts and retries.

Our nats server resides in k8s cluster, it was installed via stable helm chart.
Nats client url address is exposed via a k8s service address `nats://nats-client.default.svc.cluster.local:4222`.

**Nats server: docker.io/bitnami/nats:2.1.4-debian-10-r24**
**NUGET nats.net: 0.10.0**

Is there a way to trace the issue in the client? 
Since the golang package has never had any issues connecting to the nats server, we assume it is related to the C# client.

Thanks,
Shay

**Comment by ColinSullivan1** [comment]
@asdonation , thank you for raising this issue!  That's a fair assumption....   And Kubernetes does add layers of networking that can create issues like these.

First, have you tried increasing the [Options.Timeout](https://nats-io.github.io/nats.net/class_n_a_t_s_1_1_client_1_1_options.html#a20342f2f3c596083d5989d12ee656155) which sets the initial connect timeout?  The default is 2 seconds, maybe try something like 20 seconds (20000 millis) just to see if increasing the connect timeout resolves this.

Another place to check would be taking a look at the server logs to indicate an error.

For additional information from the server you can enable debug tracing in the server to see if there's anything reported. You can do that in [config](https://docs.nats.io/nats-tools/nas/nas_conf#logging-options) file by adding a `debug: true` line into your config or via [command line](https://docs.nats.io/nats-server/configuration/logging#command-line-options) by adding a the `-D` parameter to your NATS docker parameters, or assuming this is the right helm chart (there are a few out there), modifying [this line](https://github.com/helm/charts/blob/master/stable/nats/values.yaml#L151).

If we still can't garner the information needed (or address the issue) after these steps then we can look at instrumenting code to help debug this and/or add an inner exception to the `NATSNoServersException`.

**Comment by asdonation** [comment]
We tried raising `Options.Timeout` to 5000 ms and for now the issue does not reproduce on k8s.

Closing the issue for now,
Thanks.

**Comment by ColinSullivan1** [comment]
We're glad that resolved your issue!  Thanks for letting us know.
